base/linalg/transpose.jl

```diff
@@ -0,0 +1,334 @@
+immutable Transpose{T,S<:AbstractMatrix} <: AbstractMatrix{T}
+    data::S
+    conjugated::Bool
+end
```

Comment:
I know that I use non-conjugated transposes.   Complex-symmetric (not Hermitian) matrices come up a lot in lossy wave-propagation (Helmholtz) problems, and in these cases you end up using non-conjugated transposes.